<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LineAlignment_Widget</class>
 <widget class="QWidget" name="LineAlignment_Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QLabel" name="titleLabel">
     <property name="text">
      <string>Line Alignment to Bright Objects</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
     <property name="font">
      <font>
       <pointsize>12</pointsize>
       <bold>true</bold>
      </font>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="descriptionLabel">
     <property name="text">
      <string>Aligns line vertices to bright linear objects in media data using FWHM analysis. The algorithm calculates perpendicular directions at each vertex and finds the optimal displacement to align with bright features in the image.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="parametersGroupBox">
     <property name="title">
      <string>Alignment Parameters</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="mediaDataKeyLabel">
        <property name="text">
         <string>Media Data Key:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="mediaDataKeyComboBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="widthLabel">
        <property name="text">
         <string>Width: 20 pixels</string>
        </property>
       </widget>
      </item>
             <item row="1" column="1">
        <widget class="QSpinBox" name="widthSpinBox">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>20</number>
         </property>
         <property name="suffix">
          <string> pixels</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="perpendicularRangeLabel">
         <property name="text">
          <string>Perpendicular Range: 50 pixels</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QSpinBox" name="perpendicularRangeSpinBox">
         <property name="minimum">
          <number>10</number>
         </property>
         <property name="maximum">
          <number>200</number>
         </property>
         <property name="value">
          <number>50</number>
         </property>
         <property name="suffix">
          <string> pixels</string>
         </property>
        </widget>
       </item>
             <item row="3" column="0">
        <widget class="QLabel" name="dataTypeLabel">
         <property name="text">
          <string>Data Type:</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QCheckBox" name="useProcessedDataCheckBox">
         <property name="text">
          <string>Use processed data</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="approachLabel">
         <property name="text">
          <string>FWHM Approach:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QComboBox" name="approachComboBox"/>
       </item>
               <item row="6" column="0">
         <widget class="QLabel" name="outputModeLabel">
          <property name="text">
           <string>Output Mode:</string>
          </property>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QComboBox" name="outputModeComboBox"/>
        </item>
        <item row="7" column="0">
         <widget class="QCheckBox" name="enableGroupingCheckBox">
          <property name="text">
           <string>Enable Grouping</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="7" column="1">
         <widget class="QLabel" name="groupingDescriptionLabel">
          <property name="text">
           <string>Group FWHM lines by vertex index</string>
          </property>
          <property name="styleSheet">
           <string notr="true">color: gray;</string>
          </property>
         </widget>
        </item>
        <item row="8" column="0">
         <widget class="QLabel" name="groupPrefixLabel">
          <property name="text">
           <string>Group Prefix:</string>
          </property>
         </widget>
        </item>
        <item row="8" column="1">
         <widget class="QLineEdit" name="groupPrefixLineEdit">
          <property name="text">
           <string>FWHM_Vertex_</string>
          </property>
         </widget>
        </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="approachDescriptionLabel">
     <property name="text">
      <string>Find the width of the peak at half maximum intensity</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="styleSheet">
      <string notr="true">color: gray;</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
